dmoj/judge.py

```diff
@@ -161,6 +163,42 @@ def _block_and_grade(self, problem, language, source, short_circuit, report=prin
         else:
             self.packet_manager.grading_end_packet()
 
+    def _run_grading_wrapper_process(
```

Comment:
Maybe use the prctl here to suicide when the parent dies?

Reply:
Opted against this in the end so that we could do some best-effort reporting to show what went wrong. Any active submission will be prctl'd, though.